modules/phase_field/test/src/materials/EBCoupledVarTest.C

```diff
@@ -0,0 +1,42 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "EBCoupledVarTest.h"
+
+registerMooseObject("PhaseFieldApp", EBCoupledVarTest);
+
+template <>
+InputParameters
+validParams<EBCoupledVarTest>()
+{
+  InputParameters params = validParams<DerivativeParsedMaterialHelper>();
+  params.addClassDescription(
+      "Test to see if vector of coupled variables works with ExpressionBuilder");
+  params.addRequiredCoupledVarWithAutoBuild(
+      "v", "var_name_base", "op_num", "Array of coupled variables");
+
+  return params;
+}
+
+EBCoupledVarTest::EBCoupledVarTest(const InputParameters & parameters)
+  : DerivativeParsedMaterialHelper(parameters), _op_num(coupledComponents("v")), _vals(_op_num)
+{
+  EBTerm newest("v");
+  std::string new_name;
+  for (unsigned int i = 0; i < _op_num; ++i)
+  {
+    new_name = "v" + std::to_string(i);
+    EBTerm new_term(new_name.c_str());
+    _vals[i] = new_term;
+  }
```

Comment:
This is fine for now, but it'd be nice to have some helper method for it.

You could provide an optional `MooseObject *` argument to the `ExpressionBuilder` class. If provides through that pointer you could access the input parameters and with them their vector length. Then you could create an `EBTermVector` type with a constructor that take the input paraemeter name and sets the entire list up automatically. I.e.

```
EBTermVector vals("v");
```

instead of the loop above.